### PR TITLE
Add support for slack channel resource

### PR DESCRIPTION
**Pull request by skarimo**

### CODE REVIEW

File: datadog/resource_datadog_integration_slack_channel.go [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+	tfChannelDisplay := map[string]interface{}{}
+	tfChannelDisplay["message"] = ddChannelDisplay.GetMessage()
+	tfChannelDisplay["notified"] = ddChannelDisplay.GetNotified()
+	tfChannelDisplay["snapshot"] = ddChannelDisplay.GetSnapshot()
+	tfChannelDisplay["tags"] = ddChannelDisplay.GetTags()
+
+	return tfChannelDisplay
+}
+
+func teamNameAndChannelNameFromID(id string) (string, string, error) {
```
Comment (nmuesch) [inline datadog/resource_datadog_integration_slack_channel.go:194]: Would be good to add a unit test for this and maybe move to the internal/utils.go and internal/utils_test.go files?

File: datadog/tests/resource_datadog_integration_slack_channel_test.go [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+	accProvider := testAccProvider(t, accProviders)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    accProviders,
+		CheckDestroy: testAccCheckDatadogIntegrationSlackChannelDestroy(accProvider),
+		Steps: []resource.TestStep{
+			{
+				// Workaround to ensure we create the slack integration before running the tests.
+				Config: emptyLogsArchiveConfig(),
```
Comment (nmuesch) [inline datadog/tests/resource_datadog_integration_slack_channel_test.go:30]: nit, can we use a datasource here instead. Feels generally safer, though logs_archive_order may be fine.

Comment (nmuesch) [reply datadog/tests/resource_datadog_integration_slack_channel_test.go:30]: Could we move this method call to the PreCheck instead actually?

**Review by nmuesch — COMMENTED** [review]
Left a first pass, looking good though. 
Also please add the relevant doc pieces to the `examples` folder and run `tfplugindocs` to generate documentation

File: datadog/tests/resource_datadog_integration_slack_channel_test.go (hunk 2 above)
Comment (skarimo) [reply datadog/tests/resource_datadog_integration_slack_channel_test.go:30]: I believe the provider is initialized in the steps which is performed after the PreCheck so we might not have access to the datadog provider. Will double check

Comment (skarimo) [reply datadog/tests/resource_datadog_integration_slack_channel_test.go:30]: Good idea on the data source but unfortunately we can't use `PreCheck` in this case. It seems PreCheck happens before the provider is resolved.

Comment (jirikuncar) [reply datadog/tests/resource_datadog_integration_slack_channel_test.go:30]: You should be able to pass the `accProvider` to the `PreCheck` function via closure.

Comment (skarimo) [reply datadog/tests/resource_datadog_integration_slack_channel_test.go:30]: Gave the above a shot and ran into: 
```
panic: interface conversion: interface {} is nil, not *datadog.ProviderConfiguration [recovered]
```
Although we initialize `accProvider`, it seems that the actual provider are loaded in during the test steps only. The accProvider seems to be just a map that is used to check what providers will be used in the tests, e.g. `datadog`.

Looking at the code PreCheck itself seems to be run before the providers are resolved: https://github.com/hashicorp/terraform-plugin-sdk/blob/master/helper/resource/testing.go#L542-L544

Based on what I can tell, within Config step is the first place where the provider gets loaded in

Comment (zippolyte) [reply datadog/tests/resource_datadog_integration_slack_channel_test.go:30]: Yeah precheck is a no go, I tried it too a while ago and couldn't haha.
Do you have to use an actual resource though ? What happens if you provide an empty string ?

Comment (skarimo) [reply datadog/tests/resource_datadog_integration_slack_channel_test.go:30]: Tested it out and empty string will not work. Looks like terraform sdk expects strictly a resource in the config to load in the provider

**Review by zippolyte — APPROVED** [review]

**Pull request merged**